core/src/test/java/io/questdb/griffin/engine/groupby/TimestampSamplerFactoryTest.java

```diff
@@ -81,26 +57,83 @@ public void testNoQualifier() {
         assertFailure(100, "expected interval qualifier", "45", 98);
     }
 
+    @Test
+    public void testMinutes() throws NumericException, SqlException {
+        final StringSink sink = new StringSink();
+        final long ts = TimestampFormatUtils.parseUTCTimestamp("2022-04-23T10:33:00.123456Z");
+        for (int k = 0; k < 61; k++) {
+            final TimestampSampler sampler = createTimestampSampler(k, 'm', sink);
+            final long bucketSize = 60 * (k == 0 ? 1 : k);
```

Comment:
afaik bucket size should match the value given to the constructor in `TimestampSamplerFactory.createTimestampSampler`, which for this case is:

```
final long bucketSize = Timestamps.MINUTE_MICROS * (k == 0 ? 1 : k);
```

Reply:
So can I try like this? Thanks!
```
final long bucketSize = Timestamps.SECOND_MICROS * (k == 0 ? 1 : k);
final long expectedTs = ts - ts % bucketSize;
for (int i = 0; i < bucketSize; i+=4 * Timestamps.MILLI_MICROS) {
     long actualTs = sampler.round(expectedTs + I);
}
```